TACC/Core-CMS
# Task/gh 148 create and document element stylesheets

# Overview

Define and document stylesheets for HTML standard elements and TACC custom elements.

# Issues

- https://github.com/TACC/Core-CMS/issues/148

This will also allow proper styling of elements for hardcoded pages from [FP-893](https://jira.tacc.utexas.edu/browse/FP-893).

# Changes

- __New__: Document TACC custom element styles.
- __New__: Migrate (the few) existing HTML element styles into one stylesheet.
- __Fix__: (Noop) Typo in TACC search bar stylesheet.

# Testing Steps

1. (Optional) For new CMS installs, follow [`README.md`][readme].
1. For existing CMS installs, follow [`README.md` Quick Start][readme-quick-start], where step 1 is performed by checking out the branch.
1. Confirm that previous element styles are _still_ applied:

## Any Page
```
html,
body {
  height: 100%;
}

body {
  font-family: var(--global-font-family);
}
```

[readme]: https://github.com/TACC/Core-CMS/blob/main/README.md
[readme-quick-start]: https://github.com/TACC/Core-CMS/blob/main/README.md#quick-start

# Notes

## Bootstrap Element Styles

Bootstrap introduces many element styles before these Core CMS styles take effect.

- If we rely on Bootstrap, we should [theme it](https://getbootstrap.com/docs/4.0/getting-started/theming/) (using SASS), not overwrite it.
- But, @tacc-wbomar would rather replace Bootstrap (with in-house styles) and SASS (with [postcss-preset-env](https://preset-env.cssdb.org/)) than theme it.
- @tacc-wbomar does not know how to solve this dilemma, yet.

## (Future Task) To Do

- Migrate any one custom element's template and stylesheet and javascript into a single modular directory.
